oak_client/src/attestation.rs

```diff
@@ -109,50 +134,109 @@ impl rustls::ServerCertVerifier for RemoteAttestationVerifier {
-        })?;
-
-        // Extract X.509 TEE extension.
-        let extensions = parsed_certificate.extensions();
-        let tee_extension = extensions.get(&TEE_EXTENSION_OID).ok_or_else(|| {
-            rustls::TLSError::General("Couldn't find X.509 TEE extension".to_string())
-        })?;
-        let tee_extension_value = parse_der(tee_extension.value)
-            .map_err(|error| {
-                rustls::TLSError::General(format!(
-                    "Couldn't parse X.509 TEE extension: {:?}",
+        for certificate in certs.iter() {
+            // Parse X.509 certificate encoded using DER format.
+            // https://tools.ietf.org/html/rfc7468
+            let (_, parsed_certificate) =
+                parse_x509_certificate(&certificate.0).map_err(|error| {
+                    let error =
+                        rustls::TLSError::General(format!("Couldn't parse certificate: {}", error));
+                    warn!("{:?}", error);
                     error
-                ))
-            })?
-            .1
-            .as_str()
-            .map_err(|error| {
-                rustls::TLSError::General(format!("Couldn't parse TEE report value: {:?}", error))
-            })?;
+                })?;
 
-        // Check that X.509 TEE extension contains a correct TEE report.
-        self.check_tee_report(tee_extension_value)
-            .map_err(|error| {
-                rustls::TLSError::General(format!("Incorrect TEE report: {:?}", error))
+            // Extract public key.
+            let parsed_openssl_certificate = openssl::x509::X509::from_der(&certificate.0)
+                .map_err(|error| {
+                    let error =
+                        rustls::TLSError::General(format!("Couldn't parse certificate: {}", error));
+                    warn!("{:?}", error);
+                    error
+                })?;
+            let public_key = parsed_openssl_certificate
+                .public_key()
+                .map_err(|error| {
+                    let error = rustls::TLSError::General(format!(
+                        "Couldn't extract public key: {}",
+                        error
+                    ));
+                    warn!("{:?}", error);
+                    error
+                })?
+                .public_key_to_pem()
+                .map_err(|error| {
+                    let error = rustls::TLSError::General(format!(
+                        "Couldn't convert public key to PEM: {}",
+                        error
+                    ));
+                    warn!("{:?}", error);
+                    error
+                })?;
+
+            // Extract X.509 TEE extension.
+            let extensions = parsed_certificate.extensions();
+            let tee_extension = extensions.get(&TEE_EXTENSION_OID).ok_or_else(|| {
+                rustls::TLSError::General("Couldn't find X.509 TEE extension".to_string())
```

Comment:
Also print the id of the required extension, and the found extensions (if easy enough, if not then don't worry)